Failed to build tarantool on centos from source

**Bug description**

I tried to build Tarantool from the source files using Docker to check if the build instructions were up to date.  I ran into an error when trying to build Tarantool on CentOS 7

* OS: Centos 7

**Steps to reproduce**

```
$ docker run -ti --rm centos:7 bash

$ yum install python-pip 
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: mirror.logol.ru
 * extras: mirror.logol.ru
 * updates: mirror.logol.ru
base                                                                                                                                               | 3.6 kB  00:00:00     
extras                                                                                                                                             | 2.9 kB  00:00:00     
updates                                                                                                                                            | 2.9 kB  00:00:00     
(1/4): base/7/x86_64/group_gz                                                                                                                      | 153 kB  00:00:00     
(2/4): extras/7/x86_64/primary_db                                                                                                                  | 230 kB  00:00:00     
(3/4): updates/7/x86_64/primary_db                                                                                                                 | 6.5 MB  00:00:00     
(4/4): base/7/x86_64/primary_db                                                                                                                    | 6.1 MB  00:00:03     
No package python-pip available.
Error: Nothing to do

$ yum install epel-release
....
Installed:
  epel-release.noarch 0:7-11                                                                                                                                              

Complete!

$ yum install -y git gcc cmake3 make gcc-c++ zlib-devel readline-devel \
> ncurses-devel openssl-devel libunwind-devel libicu-devel \
> python3-yaml python3-six python3-gevent
..........
Complete!

$ git clone https://github.com/tarantool/tarantool.git --recursive

$ cd tarantool
$ git submodule update --init --recursive
$ mkdir build && cd build
[root@451ca04e6571 build]# cmake3 .. -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

<details> 
  <summary> OUTPUT </summary>
 
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Searching 16 bit integer - Using unsigned short
-- Check if the system is big endian - little endian
-- Building for Linux
-- Performing Test HAVE_STD_C11
-- Performing Test HAVE_STD_C11 - Success
-- Performing Test HAVE_STD_GNU99
-- Performing Test HAVE_STD_GNU99 - Success
-- Performing Test HAVE_STD_CXX11
-- Performing Test HAVE_STD_CXX11 - Success
-- Performing Test HAVE_STD_GNUXX0X
-- Performing Test HAVE_STD_GNUXX0X - Success
-- Performing Test HAVE_OPENMP
-- Performing Test HAVE_OPENMP - Success
-- Performing Test CC_HAS_WNO_PARENTHESES
-- Performing Test CC_HAS_WNO_PARENTHESES - Success
-- Performing Test CC_HAS_WNO_PARENTHESES_EQUALITY
-- Performing Test CC_HAS_WNO_PARENTHESES_EQUALITY - Success
-- Performing Test CC_HAS_WNO_TAUTOLOGICAL_COMPARE
-- Performing Test CC_HAS_WNO_TAUTOLOGICAL_COMPARE - Success
-- Performing Test CC_HAS_WNO_MISLEADING_INDENTATION
-- Performing Test CC_HAS_WNO_MISLEADING_INDENTATION - Success
-- Performing Test CC_HAS_WNO_VARARGS
-- Performing Test CC_HAS_WNO_VARARGS - Success
-- Performing Test CC_HAS_WNO_CHAR_SUBSCRIPTS
-- Performing Test CC_HAS_WNO_CHAR_SUBSCRIPTS - Success
-- Performing Test CC_HAS_WNO_FORMAT_TRUNCATION
-- Performing Test CC_HAS_WNO_FORMAT_TRUNCATION - Success
-- Performing Test CC_HAS_WNO_IMPLICIT_FALLTHROUGH
-- Performing Test CC_HAS_WNO_IMPLICIT_FALLTHROUGH - Success
-- Performing Test CC_HAS_WNO_CAST_FUNCTION_TYPE
-- Performing Test CC_HAS_WNO_CAST_FUNCTION_TYPE - Success
-- Performing Test CC_HAS_GGDB
-- Performing Test CC_HAS_GGDB - Success
-- Looking for libunwind.h
-- Looking for libunwind.h - /usr/include
-- Looking for libunwind.h
-- Looking for libunwind.h - found
-- Found unwind
-- Enabling LTO: FALSE
-- Performing Test CC_HAS_SSE2_INTRINSICS
-- Performing Test CC_HAS_SSE2_INTRINSICS - Success
-- Performing Test CC_HAS_AVX_INTRINSICS
-- Performing Test CC_HAS_AVX_INTRINSICS - Success
-- SSE2 is enabled - target CPU must supppot it
-- Performing Test CC_HAS_ATOMICS
-- Performing Test CC_HAS_ATOMICS - Success
-- Looking for __gcov_flush in 
-- Looking for __gcov_flush in  - found
-- Looking for pthread_np.h
-- Looking for pthread_np.h - not found
-- Performing Test HAVE_PTHREAD_SETNAME_NP
-- Performing Test HAVE_PTHREAD_SETNAME_NP - Success
-- Performing Test HAVE_PTHREAD_SETNAME_NP_1
-- Performing Test HAVE_PTHREAD_SETNAME_NP_1 - Failed
-- Performing Test HAVE_PTHREAD_SET_NAME_NP
-- Performing Test HAVE_PTHREAD_SET_NAME_NP - Failed
-- Performing Test HAVE_PTHREAD_GETATTR_NP
-- Performing Test HAVE_PTHREAD_GETATTR_NP - Success
-- Performing Test HAVE_PTHREAD_STACKSEG_NP
-- Performing Test HAVE_PTHREAD_STACKSEG_NP - Failed
-- Performing Test HAVE_PTHREAD_ATTR_GET_NP
-- Performing Test HAVE_PTHREAD_ATTR_GET_NP - Failed
-- Performing Test HAVE_PTHREAD_GET_STACKSIZE_NP
-- Performing Test HAVE_PTHREAD_GET_STACKSIZE_NP - Failed
-- Performing Test HAVE_PTHREAD_GET_STACKADDR_NP
-- Performing Test HAVE_PTHREAD_GET_STACKADDR_NP - Failed
-- Looking for MAP_ANON
-- Looking for MAP_ANON - found
-- Looking for MAP_ANONYMOUS
-- Looking for MAP_ANONYMOUS - found
-- Looking for MADV_DONTNEED
-- Looking for MADV_DONTNEED - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for cpuid.h
-- Looking for cpuid.h - found
-- Looking for sys/prctl.h
-- Looking for sys/prctl.h - found
-- Looking for O_DSYNC
-- Looking for O_DSYNC - found
-- Looking for fdatasync
-- Looking for fdatasync - found
-- Looking for pthread_yield
-- Looking for pthread_yield - found
-- Looking for sched_yield
-- Looking for sched_yield - found
-- Looking for posix_fadvise
-- Looking for posix_fadvise - found
-- Looking for fallocate
-- Looking for fallocate - found
-- Looking for mremap
-- Looking for mremap - found
-- Looking for sync_file_range
-- Looking for sync_file_range - found
-- Looking for memmem
-- Looking for memmem - found
-- Looking for memrchr
-- Looking for memrchr - found
-- Looking for sendfile
-- Looking for sendfile - found
-- Looking for uuidgen
-- Looking for uuidgen - not found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for __get_cpuid
-- Looking for __get_cpuid - found
-- Looking for strlcpy
-- Looking for strlcpy - not found
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM - Success
-- Performing Test HAVE_STRUCT_STAT_ST_MTIMENSEC
-- Performing Test HAVE_STRUCT_STAT_ST_MTIMENSEC - Failed
-- Looking for __libc_stack_end in 
-- Looking for __libc_stack_end in  - found
-- Looking for setproctitle
-- Looking for setproctitle - not found
-- Looking for setprogname
-- Looking for setprogname - not found
-- Looking for getprogname
-- Looking for getprogname - not found
-- Generating VERSION file
-- Updating submodules
-- Tarantool version is 2.8.0-173-g83ec719 (2.8.0.173)
-- Lua package.path: /usr/local/share/tarantool/?.lua;/usr/local/share/tarantool/?/init.lua;/usr/share/tarantool/?.lua;/usr/share/tarantool/?/init.lua;/usr/local/share/lua/5.1/?.lua;/usr/local/share/lua/5.1/?/init.lua;/usr/share/lua/5.1/?.lua;/usr/share/lua/5.1/?/init.lua
-- Lua package.cpath: /usr/local/lib64/tarantool/?.so;/usr/lib64/tarantool/?.so;/usr/local/lib64/lua/5.1/?.so;/usr/lib64/lua/5.1/?.so
-- Using bundled ZSTD
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.0.2k")  
-- OpenSSL 1.0.2k found
-- Looking for zlib
-- Looking for zlib.h - /usr/include
-- Looking for libz - /usr/lib64/libz.so
-- EXPORT_LIBCURL_SYMBOLS: ON
-- Found Curses: /usr/lib64/libcurses.so  
-- Found Readline: /usr/include  
-- Looking for rl_catch_sigwinch in /usr/lib64/libreadline.so
-- Looking for rl_catch_sigwinch in /usr/lib64/libreadline.so - found
-- Detected GNU Readline
-- Performing Test ICONV_RUNS
-- Performing Test ICONV_RUNS - Success
-- Found ICU: /usr/include  
-- Performing Test HAVE_ICU_STRCOLLUTF8
-- Performing Test HAVE_ICU_STRCOLLUTF8 - Success
-- [SetVersion] Reading version from VCS: v2.1.0-beta3-89-gcdf98db
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc

CMake Warning at third_party/luajit/test/lua-Harness-tests/CMakeLists.txt:9 (message):
  `prove' is not found, so lua-Harness-tests target is not generated

CMake Warning at third_party/luajit/test/tarantool-tests/CMakeLists.txt:9 (message):
  `prove' is not found, so tarantool-tests target is not generated

-- Using bundled MsgPuck
-- Using bundled libyaml
-- Looking for MAP_ANON
-- Looking for MAP_ANON - found
-- Looking for MAP_ANONYMOUS
-- Looking for MAP_ANONYMOUS - found
-- Looking for madvise
-- Looking for madvise - found
-- Looking for MADV_DONTDUMP
-- Looking for MADV_DONTDUMP - found
\-- 
-- Performing Test CC_HAS_MNO_UNALIGNED_ACCESS
-- Performing Test CC_HAS_MNO_UNALIGNED_ACCESS - Failed
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test HAVE_MSG_NOSIGNAL
-- Performing Test HAVE_MSG_NOSIGNAL - Success
-- Performing Test HAVE_SO_NOSIGPIPE
-- Performing Test HAVE_SO_NOSIGPIPE - Failed
\-- 
-- PACKAGE: Tarantool
-- VERSION: 2.8.0-173-g83ec719
-- BUILD: Linux-x86_64-RelWithDebInfo
-- C_COMPILER: /usr/bin/cc
-- CXX_COMPILER: /usr/bin/c++
-- C_FLAGS:  -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type
-- CXX_FLAGS:  -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type
-- PREFIX: /usr/local
-- ENABLE_SSE2: ON
-- ENABLE_AVX: OFF
-- ENABLE_GCOV: OFF
-- ENABLE_GPROF: OFF
-- ENABLE_VALGRIND: OFF
-- ENABLE_ASAN: OFF
-- ENABLE_UB_SANITIZER: OFF
-- ENABLE_FUZZER: OFF
-- ENABLE_BACKTRACE: ON
-- ENABLE_DIST: OFF
-- ENABLE_BUNDLED_LIBCURL: ON
-- ENABLE_BUNDLED_LIBYAML: ON
-- ENABLE_BUNDLED_MSGPUCK: ON
\-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /tarantool/build
</details>

```
$ [root@451ca04e6571 build]# make 
   [  1%] Built target decNumber
   [  1%] Built target eio
   [  1%] Built target coro
   [  1%] Built target misc
   [  2%] Built target bundled-ares-project
   [  2%] Performing configure step for 'bundled-libcurl-project'
   /bin/sh: cmake: command not found
   make[2]: *** [build/curl/work/stamp/bundled-libcurl-project-configure] Error 127
   make[1]: *** [CMakeFiles/bundled-libcurl-project.dir/all] Error 2
   make: *** [all] Error 2

```

--- a/changelogs/unreleased/fix-cmake3-build.md
+++ b/changelogs/unreleased/fix-cmake3-build.md
@@ -1,0 +1,7 @@
+## bugfix/build
+
+* Fixed libcurl configuring, when tarantool itself is configured with `cmake3`
+  command and there is no `cmake` command in PATH (gh-5955).
+
+  This affects building tarantool from sources with bundled libcurl (it is the
+  default mode).

--- a/cmake/BuildLibCURL.cmake
+++ b/cmake/BuildLibCURL.cmake
@@ -154,7 +154,7 @@ macro(curl_build)
         STAMP_DIR ${LIBCURL_BINARY_DIR}/stamp
         BINARY_DIR ${LIBCURL_BINARY_DIR}/curl
         CONFIGURE_COMMAND
-            cd <BINARY_DIR> && cmake <SOURCE_DIR>
+            cd <BINARY_DIR> && ${CMAKE_COMMAND} <SOURCE_DIR>
                 ${LIBCURL_CMAKE_FLAGS}
         BUILD_COMMAND cd <BINARY_DIR> && $(MAKE)
         INSTALL_COMMAND cd <BINARY_DIR> && $(MAKE) install)

